### ISSUE TITLE
False negative from DL3040 when using dnf upgrade

### ISSUE THREAD

**Issue by duncanmmacleod** [hadolint/hadolint#1179]
`hadolint` doesn't seem to complain about a missing `dnf clean` ([DL3040](https://github.com/hadolint/hadolint/wiki/DL3040)) after `dnf upgrade` (used to update all packages).

```dockerfile
FROM quay.io/rockylinux/rockylinux:8

RUN dnf -y upgrade
```

Is it reasonable to add support for this?

Apologies if this has already been reported/discussed.

**Comment by faust64** [comment]
as of updating to 2.15.1, I'm now getting those DL3040 errors

```
RUN set -x \
...
    && microdnf clean all \
    && microdnf update -y \
    && microdnf install -y --nodocs \
        "--enablerepo=rhel-10-for-$(uname -m)-baseos-rpms" \
        "--enablerepo=rhel-10-for-$(uname -m)-appstream-rpms" \
        [...] \
    && microdnf clean all \
    && rm -rf /var/cache/dnf /etc/pki/entitlement /etc/rhsm /usr/share/doc /usr/share/man /usr/local/share/man/
```

In a single RUN, I update/install my packages then "microdnf clean all"
We now have false positive for DL3040?

**Comment by m-ildefons** [comment]
Hi @faust64 ,

you have two instances of `microdnf clean all` and the false-positive you're seeing is happening on the first one.
This should be ok:

```dockerfile
RUN set -x \
    [...]
    && microdnf update -y \
    && microdnf install -y --nodocs \
        "--enablerepo=rhel-10-for-$(uname -m)-baseos-rpms" \
        "--enablerepo=rhel-10-for-$(uname -m)-appstream-rpms" \
    [...] \
    && microdnf clean all \
    && rm -rf /var/cache/dnf /etc/pki/entitlement /etc/rhsm /usr/share/doc /usr/share/man /usr/local/share/man/
```

**Comment by faust64** [comment]
Hello @m-ildefons 

I just tried removing that first microdnf clean all: DL3040 still fires

<img width="901" height="379" alt="Image" src="https://github.com/user-attachments/assets/f04cdbe9-fa65-40f8-9415-a94529c98129" />

**Comment by Wuodan** [comment]
The early `rm -rf /etc/rhsm-host` is treated as a cleanup for `dnf` (despite the removed folder not being related to dnf) as described by issue #1250 and fixed by PR #1251.

I did not run a test though as sadly this issue has a screenshot of a Dockerfile instead of the file or the text in the file.
